### PR TITLE
test(extension): coverage for new stake pool sorting

### PR DESCRIPTION
**Pull request by wklos-iohk**
# Checklist

- [X] JIRA - LW-10301
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

### CODE REVIEW

**Comment by wklos-iohk** [comment]
Smoke tests are failing due to unstable env

File: packages/e2e-tests/src/elements/multidelegation/MultidelegationPage.ts [hunk 1]
```diff
@@ -543,6 +587,69 @@ class MultidelegationPage {
...
+    await browser.waitUntil(async () => (await this.getNumberOfPoolsFromCounter()) > 0, {
+      timeoutMsg: 'No stake pools!'
+    });
+  }
+
+  async extractColumnContent(columnName: StakePoolListColumn, poolLimit = 100): Promise<string[]> {
+    const columnContent: string[] = [];
+
+    await this.listContainer.waitForStable();
+    await browser.pause(500);
```
Comment (oldGreg5) [inline packages/e2e-tests/src/elements/multidelegation/MultidelegationPage.ts:607]: There's no other way but infamous pause?

**Review by oldGreg5 — CHANGES_REQUESTED** [review]
Nice work on sorting bonanza ;)

File: packages/e2e-tests/src/assert/multidelegation/MultidelegationPageAssert.ts [hunk 2]
```diff
@@ -188,36 +192,37 @@ class MultidelegationPageAssert {
     expect(await firstStakePool.ticker.getText()).to.equal(expectedTicker);
   };
 
-  assertSeeTooltipForColumn = async (columnName: string) => {
+  assertSeeTooltipForColumn = async (column: StakePoolListColumn) => {
+    await MultidelegationPage.tooltip.waitForStable();
     await MultidelegationPage.tooltip.waitForDisplayed();
```
[the remark this answers is not in the document]
Comment (wklos-iohk) [reply packages/e2e-tests/src/assert/multidelegation/MultidelegationPageAssert.ts:197]: yes, they do 2 separate things

File: packages/e2e-tests/src/elements/multidelegation/MultidelegationPage.ts (hunk 1 above)
Comment (wklos-iohk) [reply packages/e2e-tests/src/elements/multidelegation/MultidelegationPage.ts:607]: Unfortunately I wasn't able to find another way, even `waitForStable()` wasn't helpful here

File: packages/e2e-tests/src/assert/multidelegation/MultidelegationPageAssert.ts [hunk 3]
```diff
@@ -300,6 +305,39 @@ class MultidelegationPageAssert {
...
+  assertSeeStakePoolsSorted = async (
+    stakePoolsDisplayType: 'list rows' | 'cards',
+    sortingOption: StakePoolListColumnName,
+    order: SortingOrder,
+    poolLimit?: number
+  ) => {
+    await MultidelegationPage.waitForPoolsCounterToBeGreaterThanZero();
+    poolLimit ??= await MultidelegationPage.getNumberOfPoolsFromCounter();
+    if (stakePoolsDisplayType === 'cards') {
+      // TODO: add code to handle grid cards
```
[the remark this answers is not in the document]
Comment (wklos-iohk) [reply packages/e2e-tests/src/assert/multidelegation/MultidelegationPageAssert.ts]: Ticket number added

File: packages/e2e-tests/src/features/MultiDelegationPageExtended.part1.feature [hunk 4]
```diff
@@ -88,22 +88,27 @@ Feature: Staking Page - Extended View
     Then I see the Network Info component with the expected content
 
   @LW-8499 @Testnet @Mainnet
-  Scenario Outline: Extended View - Staking - Show tooltip for <column_name> column in browse pools section
+  Scenario: Extended View - Staking - Show tooltip for <column_name> column in browse pools section
```
[the remark this answers is not in the document]
Comment (wklos-iohk) [reply packages/e2e-tests/src/features/MultiDelegationPageExtended.part1.feature]: updated

File: packages/e2e-tests/src/features/MultiDelegationPageExtended.part2.feature [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  Background:
+    Given Lace is ready for test
+
+  @LW-10143 @Testnet @Mainnet
+  Scenario: Extended View - Staking - More options - Sorting options are displayed
+    When I am on Staking extended page
+    And I open Browse pools tab
+    Then "More options" component with stake pool sorting options is displayed
+
+  @LW-10139 @LW-1041 @LW-10142 @Testnet @Mainnet
```
[the remark this answers is not in the document]
Comment (wklos-iohk) [reply packages/e2e-tests/src/features/MultiDelegationPageExtended.part2.feature]: updated

File: packages/e2e-tests/src/features/MultiDelegationPageExtended.part2.feature [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    Examples:
+      | column     | default_order | modified_order |
+      | Saturation | descending    | ascending      |
+#      | ROS         | descending  | ascending      |# TODO: Uncomment when USE_ROS_STAKING_COLUMN=true
+      | Cost       | ascending     | descending     |
+      | Margin     | ascending     | descending     |
+      | Blocks     | descending    | ascending      |
+      | Pledge     | descending    | ascending      |
+      | Live Stake | descending    | ascending      |
+#      | Ticker      | ascending   | descending     |# Tested in LW-10139
```
[the remark this answers is not in the document]
Comment (wklos-iohk) [reply packages/e2e-tests/src/features/MultiDelegationPageExtended.part2.feature]: removed

File: packages/e2e-tests/src/elements/multidelegation/MultidelegationPage.ts [hunk 7]
```diff
@@ -543,6 +587,69 @@ class MultidelegationPage {
     const selectedTickers = await this.getTickersOfSelectedPools(viewType);
     testContext.save('selectedTickers', selectedTickers);
   }
+
+  async getNumberOfPoolsFromCounter(): Promise<number> {
+    const poolsCounterText = await this.poolsCounter.getText();
+    return Number(poolsCounterText.replace('Pools (', '').replace(',', '').replace(')', ''));
```
[the remark this answers is not in the document]
Comment (wklos-iohk) [reply packages/e2e-tests/src/elements/multidelegation/MultidelegationPage.ts]: updated with regex covering all characters before number enclosed in `(...)`

**Review by oldGreg5 — APPROVED** [review]

**Pull request merged**